frontend/document/koptinterface.lua

```diff
@@ -1510,15 +1511,18 @@ function KoptInterface:findAllText(doc, pattern, case_insensitive, nb_context_wo
                         i_next, j_next = i, j
                     end
                 end
-                text = "【" .. table.concat(text, " ") .. "】"
+                -- Make this word bolder, using Poor Text Formatting provided by TextBoxWidget
+                -- (we know this text ends up in a TextBoxWidget).
+                text = TextBoxWidget.PTF_BOLD_START .. table.concat(text, " ") .. TextBoxWidget.PTF_BOLD_END
                 local prev_text = get_prev_text(text_boxes, i_prev, j_prev, nb_context_words)
                 if prev_text then
-                    text = prev_text .. text
+                    text = prev_text .. " " .. text
                 end
                 local next_text = get_next_text(text_boxes, i_next, j_next, nb_context_words)
                 if next_text then
-                    text = text .. next_text
+                    text = text .. " " .. next_text
```

Comment:
@hius07 : can you confirm this is alright, adding spaces here ?
It was visually needed (otherwise words get stuck), I'm just not sure if it should always be a space (probably shouldn't with CJK), and if there's something better to do (not really familiar with this kopt/pdf part).

Reply:
https://github.com/koreader/koreader-base/blob/3e94c069bb623bbf1ceac19674d910b4302f82eb/ffi/mupdf.lua#L535--L545
Word separators.
No idea what is better than adding spaces.
If CJK users complain, we will ask them for the propositions.

Reply:
Would a ZWNJ work in this context?

Reply:
Given the zero width I very much doubt it?

Reply:
Nope.
It's not a rendering issue, but about re-building the sentence from the original splitted sentece we get as a list of "words" from ffi/mupdf.lua.
I guess the pointed code could add a `word.is_cjk` so we can decide in frontend to add or not a space when one or both contiguous words are CJK.

Reply:
@hius07 : in the mupdf.lua getPageText() you linked to, there seems to me there are some logic error: `English惊` would come up as one word, and so would a word ending with any of these https://en.wikipedia.org/wiki/General_Punctuation.
I suppose it should make a word with what's before (`English`), and go on make another word with that CJK char met (`惊`).
I could try to fix that, but a question:

Do we need this function to return the same words that it had in the past, or is that ok if it does not split the same segments/words ?
Ie. do we have some code, like in ReaderBookmark/Highlight, that needs to rematch a highlight saved text to the original text ? ie. to know if a highlight already exists, or to match "annotations written in the PDF" for removal/edit?
If no, well, I'm a bit free to try to do the right things there (including adding or not spaces in the right conditions).
If yes, is the full text sufficient to be equal - that is, if words are splitted differently, but when concatenated would end up with the same text? I would then be a little less free, and should try to make the same sometimes wrong spaces as before.

Reply:
The first paragraph, about fixing CJK, may be a matter of opinion, and ok if you have none :)

But the 2nd paragraph is a matter of facts, and memory (of stuff you probably touched more recently than I).
I just remembered you not long ago told me finding annotations in PDF was using pboxes (only?), so probably not the text string.
Anyway, fine too if you have no strong memory (like me :/) :)

Reply:
You are right, finding annotations requires pboxes only
https://github.com/koreader/koreader/blob/43d36b2ea9dbe894e6aae4b0f7aca3fbe1ce369d/frontend/document/pdfdocument.lua#L272-L274

Reply:
No, and in some cases we use synthetic item
https://github.com/koreader/koreader/blob/43d36b2ea9dbe894e6aae4b0f7aca3fbe1ce369d/frontend/apps/reader/modules/readerhighlight.lua#L1891